### PR TITLE
Clean up syntax for supported Python versions.

### PR DESCRIPTION
**Pull request by alok**
## What do these changes do?

Replaces syntax with better syntax.

The dict/set/tuple literals (curly braces) for comprehensions and instantiation is
clearer and faster than writing unidiomatic calls to each of those functions.

## Related issue number

No.

### CODE REVIEW

File: python/ray/cloudpickle/cloudpickle.py [hunk 1]
```diff
@@ -572,8 +572,8 @@ def extract_code_globals(cls, co):
                 # PyPy "builtin-code" object
                 out_names = set()
             else:
-                out_names = set(names[oparg]
-                                for op, oparg in _walk_global_ops(co))
```
Comment (robertnishihara) [inline python/ray/cloudpickle/cloudpickle.py]: Let's not make any changes to this file as it is directly copied from the cloudpickle repository.

File: python/ray/dataframe/dataframe.py [hunk 2]
```diff
@@ -260,7 +260,7 @@ def __repr__(self):
         # The split here is so that we don't repr pandas row lengths.
         result = self._repr_helper_()
         final_result = repr(result).rsplit("\n\n", maxsplit=1)[0] + \
-            "\n\n[{0} rows x {1} columns]".format(len(self.index),
+            "\n\n[{} rows x {} columns]".format(len(self.index),
```
Comment (robertnishihara) [inline python/ray/dataframe/dataframe.py]: If we do this change (and others) we'll need to fix the indentation of the next line

**Comment by robertnishihara** [comment]
@devin-petersohn can you take a look at the changes to the dataframes files and see if these make sense to you? If you prefer we can make these changes only for non-dataframes files.

**Comment by devin-petersohn** [comment]
@robertnishihara @Alok These changes are fine for the dataframes files. Thanks!

File: python/ray/cloudpickle/cloudpickle.py (hunk 1 above)
Comment (Alok) [reply python/ray/cloudpickle/cloudpickle.py]: Good point. Put in a PR w/cloudpickle to change that though.

File: python/ray/dataframe/dataframe.py (hunk 2 above)
Comment (Alok) [reply python/ray/dataframe/dataframe.py]: I'm fine leaving this as is then, though I'd prefer that there be a git hook to require all code to be run through auto-formatters. It'd unify things like the use of certain syntax, indent, and quote styles.

**Comment by Alok** [comment]
I think those are both of the necessary changes.

**Comment by robertnishihara** [comment]
Looks good, thanks! There are still a couple linting errors at https://travis-ci.org/ray-project/ray/jobs/372891975. Could you fix those?

**Comment by alok** [comment]
I think that's all the lint errors taken care of.

**Comment by robertnishihara** [comment]
Thanks @alok, yapf seems to be complaining (though unfortunately it doesn't list the diff).

```
$ .travis/yapf.sh
~/build/ray-project/ray/test ~/build/ray-project/ray
~/build/ray-project/ray
~/build/ray-project/ray/python ~/build/ray-project/ray
~/build/ray-project/ray
Reformatted staged files. Please review and stage the changes.
Files updated:
  python/ray/worker.py
  test/actor_test.py
  test/stress_tests.py
```

**Comment by pcmoritz** [comment]
There are still linting errors, can you have another look?

**Comment by robertnishihara** [comment]
Maybe it's a version issue with yapf?

**Comment by alok** [comment]
The cloudpickle changes were [just merged in](https://github.com/cloudpipe/cloudpickle/pull/169#issuecomment-386170715), so we can pick them up next time we update those files.

**Comment by alok** [comment]
@robertnishihara 

[Closing and reopening a PR apparently forces Travis to rebuild](https://github.com/travis-ci/travis-ci/issues/576#issuecomment-65317915), and the lint test is green now.

**Comment by pcmoritz** [comment]
Jenkins test this please

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**